### DISCUSSION TITLE
Calling [AlsoNotifyCanExecuteFor] on a Get Only Property for an ICommand

### DISCUSSION THREAD

**Discussion by BinaryAssault** [CommunityToolkit/dotnet#185] [Q&A] [answered]
I have an ICommand that needs a CanExecute attached to it.

The CanExecute is triggered by a get only property that is updated elsewhere using [AlsoNotifyChangeFor(nameof(SomeBool)]. 

I can't attach [AlsoNotifyCanExecuteFor] to my get only property since it's not a field.

What is the correct way to go about this? I really enjoy the code generators and would like to not have to write out the relay command if I don't have to.

```csharp
[ObservableProperty]
[AlsoNotifyChangeFor(nameof(SomeBool))]
private string someString;

[AlsoNotifyCanExecuteFor(nameof(SomeFunctionCommand))] // <-- point of failure
public bool SomeBool => someClass.Status;

[ICommand]
private void SomeFunction()
{
    // allowed to do stuff iff SomeBool is true
}
```

**Comment by Sergio0694** [comment, marked as answer]
In this case you'd need to just move `[AlsoNotifyCanExecuteFor(nameof(SomeFunctionCommand)]` over `someString`. Also, note that you will also need to set `CanExecute = nameof(SomeBool)` in your command, otherwise it won't actually use the value of that property to decide whether it can be executed 🙂

**Comment by BinaryAssault** [comment]
Hi @Sergio0694, can you give me an example of 

> set CanExecute = nameof(SomeBool) in your command

I don't quite understand it. I'm sure I used it wrong

```csharp
[ICommand]
private void SomeFunction()
{ 
    CanExecute = nameof(SomeBool); // doesn't compile.
    // allowed to do stuff iff SomeBool is true
}
```

**Reply by Sergio0694** [reply]
Oh, sorry. It's a property in the command attribute, so it should just be:

```csharp
[ICommand(CanExecute = nameof(SomeBool))]
private void SomeFunction()
{
    // allowed to do stuff iff SomeBool is true
}
```

**Reply by BinaryAssault** [reply]
Ah! I knew it was something simple I was missing. Thank you!!

**Comment by BinaryAssault** [comment]
I implemented as suggested, and now I get a reflection error.  
Removing this line gets rid of the error, but obviously never updates the CanExecute
``````
[AlsoNotifyCanExecuteFor(nameof(SomeFunctionCommand))]
``````
Full example
``````
[ObservableProperty]
[AlsoNotifyChangeFor(nameof(SomeBool))]
[AlsoNotifyCanExecuteFor(nameof(SomeFunctionCommand))]
private string someString;

public bool SomeBool => someClass.Status;

[ICommand(CanExecute=nameof(SomeBool))]
private void SomeFunction()
{
    // allowed to do stuff iff SomeBool is true
}
``````

![image](https://user-images.githubusercontent.com/19739224/160865176-4d52ceca-5c4d-4de9-81ad-1d902e5fe95a.png)

**Reply by Sergio0694** [reply]
This discussion was about how to notify a command (which was answered), but it's not the right place to report a bug (or a presumed bug). Could you open a new bug report issue and follow the template to provide all required info (including a minimal repro) so we can look into it? Thank you! 🙂

**Reply by BinaryAssault** [reply]
Will do, thank you @Sergio0694 !
